cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/common/submit/PipelinePhasePreparer.java

```diff
@@ -89,7 +90,7 @@ public PipelinePhasePreparer(PluginContext pluginContext, Metrics metrics, Macro
       boolean isConnectorSink =
         Constants.Connector.PLUGIN_TYPE.equals(pluginType) && phase.getSinks().contains(stageName);
 
-      SubmitterPlugin submitterPlugin;
+      SubmitterPlugin submitterPlugin = null;
```

Comment:
So the `submitterPlugin` can be `null` after the following `if`s checks? Is it being handled already? I don't think the logic can this value being `null`. It is probably better to have a `else` that throw if the joiner plugin is not of one of the supported classes.